### PR TITLE
Update DG Wk 9 Tutorial

### PR DESCRIPTION
**Pull request by jaivigneshvenugopal**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -345,6 +345,60 @@ image::UndoRedoActivityDiagram.png[width="200"]
 **Cons:** Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things. +
 // end::undoredo[]
 
+// tag::list[]
+=== List (Mainlist, Blacklist etc) Mechanism
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: maybe you can change 'List (Mainlist, Blacklist etc) mechanism'

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -345,6 +345,60 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+
+List feature is a useful feature to view persons of a certain category. In the context of this project, master list contains all the contacts of the debt collection agency. As the need to categorize people into the respective groups was necessary, we implemented a new way to obtain this.
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: I think you should keep your explanation solely to just describing the mechanism itself. How useful the feature is not important in this case. Keep it concise and remove phrases that do not add value to what you are trying to explain

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -345,6 +345,60 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+
+We are displaying multiple lists to suit the need for the direction of the project. The lists could be viewed with the respective commands that will bring it to picture. The commands to materialize these lists are named after the lists itself. Meaning, to view the blacklist, the command typed is "blacklist" or "bl'.
+
+These commands will inherit from `UndoableCommand`. Thus these commands are supported by #undo and `redo` command residing in the current state of Addressbook.
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: - Thus, *
- is '#undo' intentional? or was it supposed to be`undo`?

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -345,6 +345,60 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+
+This List mechanism is facilitated by commands which uses `Logic` `Interface` to obtain the copy of the list that is required. As seen by the picture above, to obtain the current blacklist of the address book, "list" string is first captured by the `CommandBox`. The `CommandBox` then passes this string to the Logic interface for execution.
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: - Perhaps refer to the figure by its figure number.
- As seen from*
- `Logic` interface or `Logic` `Interface`?

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -345,6 +345,60 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+
+`Logic` `Interface` uses `LogicManager`, the class that implements its methods, to package it as a `Command` object. The respective command is then executed by this following statement in `LogicManager` class:
+
+CommandResult result = command.execute();
```
Comment (jelneo) [inline docs/DeveloperGuide.adoc]: use this tag for code from source:

[source,java]
----
----

**Review by jelneo — APPROVED** [review]
Some minor issues. Good job :)

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -345,6 +345,60 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+
+==== Design Constraints:
+
+**Aspect:** Other commands still work in Blacklist mode(`Add`, `Delete` command). +
+**Alternative 1 (current choice):** Inform user in user guide to use only specific commands for that list. +
+**Pros:** Easy to implement. +
+**Cons:** Performance issues as users are bound to make a mistake. +
+**Alternative 2:** Restrict other command usages in certain lists. +
+**Pros:** Users will never make a mistake sending the wrong person to the wrong list. +
+**Cons:** Implementation is complicated and ever changing whenever new commands are introduced.
```
Comment (khooroko) [inline docs/DeveloperGuide.adoc]: what is ever changing wth

File: docs/DeveloperGuide.adoc [hunk 7]
```diff
@@ -345,6 +345,60 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+
+`Logic` `Interface` uses `LogicManager`, the class that implements its methods, to package it as a `Command` object. The respective command is then executed by this following statement in `LogicManager` class:
+
+CommandResult result = command.execute();
+
+Specifically for `blacklist` command, it calls the `getFiltererdBlacklistedPersons()` method residing in the `Logic` `Interface`. `Logic` `Manager` subsequently calls `Model` `Interface`. `ModelManager` handles the command and thereafter calls `ReadOnlyAddressBook` `Interface` to handle the request. `AddressBook` class handles this request and thus will return the required list, since the various lists are stored in this class. Technically, the `List` `Object` that exists in the current structure just contains all persons. The blacklisted persons are obtained by running a check on all persons residing in this variable. This is shown by the following code below:
```
Comment (khooroko) [inline docs/DeveloperGuide.adoc]: `LogicManager`* not seperate

File: docs/DeveloperGuide.adoc [hunk 8]
```diff
@@ -345,6 +345,60 @@ image::UndoRedoActivityDiagram.png[width="200"]
...
+
+[source,java]
+----
+CommandResult result = command.execute();
+----
+
+Specifically for `blacklist` command, it calls the `getFiltererdBlacklistedPersons()` method residing in the `Logic` `Interface`. `Logic` `Manager` subsequently calls `Model` `Interface`. `ModelManager` handles the command and thereafter calls `ReadOnlyAddressBook` `Interface` to handle the request. `AddressBook` class handles this request and thus will return the required list, since the various lists are stored in this class. Technically, the `List` `Object` that exists in the current structure just contains all persons. The blacklisted persons are obtained by running a check on all persons residing in this variable. This is shown by the following code below:
```
Comment (khooroko) [inline docs/DeveloperGuide.adoc]: why do you have 2 of the same line

File: docs/DeveloperGuide.adoc (hunk 6 above)
Comment (jaivigneshvenugopal) [reply docs/DeveloperGuide.adoc]: Noted about the error

**Pull request merged**